storage_controller/src/persistence.rs

```diff
@@ -572,6 +572,30 @@ impl Persistence {
             .collect())
     }
 
+    /// Read the generation number of a specific tenant shard
+    pub(crate) async fn get_generation(
+        &self,
+        tenant_shard_id: TenantShardId,
+    ) -> Result<Option<Generation>, DatabaseError> {
+        use crate::schema::tenant_shards::dsl::*;
+        let mut rows = self
+            .with_measured_conn(DatabaseOperation::PeekGenerations, move |conn| {
```

Comment:
Is it common to use a `DatabaseOperation` variant in multiple helpers? because here, we do.